src/main/scala/state/PreconditionPropagationTransformer.scala

```diff
@@ -0,0 +1,36 @@
+package viper.silicon.state
+
+import viper.silicon.rules.functionSupporter
+import viper.silicon.state.terms.{And, App, HeapDepFun, Implies, Ite, Let, Literal, Not, Or, Quantification, Term, True}
+import viper.silver.ast
+
+object PreconditionPropagationTransformer {
```

Comment:
Assuming the transformer propagates preconditions, how about `PreconditionPropagatingTransformer`?

Reply:
After writing down what it actually does, I settled on `FunctionPreconditionTransformer`. While it doesn't transform function preconditions, it transforms terms to state that all function preconditions are fulfilled.